server/tests/endpoints.test.js

```diff
@@ -0,0 +1,81 @@
+import 'babel-polyfill'
+import axios from 'axios'
+import { describe, expect, jest, test } from '@jest/globals'
+import { databaseApi, searchApi } from '../endpoints'
+import { articles } from './mocks'
+import { DATABASE_ENDPOINT, SEARCH_ENDPOINT } from '../config'
+
+jest.mock('axios')
+
+const searchApiTests = [
+  {
```

Comment:
Yeah, technically they are there to verify that the **mocked services** return the appropriate payload. And what you say makes sense as in if the external service changes its payload format, these unit tests won't actually break.

I added them as per @ssmall suggestion. Spencer, is there any other consideration as to why testing the mocked endpoints is important?

Reply:
> Do any of these tests run through any additional logic from your frontend/backend code? Or only mocks for external service? 

The logic that they test is very concise (because Javascript), but I think it's valuable -- it's testing that the `searchApi` function does actually attempt to make a `post` request to the appropriate URL. Without this test, `searchApi` could be redefined to just `return undefined` and all other unit tests would still pass.

Reply:
Ah, ok, that's fair. I think testing the URL is a reasonable test, but is better done with just one test case, and the `expected` field of the test case should be the URL expected in that case.

I would say `query` and `response` fields can be dropped from the `searchApiTests` elements, and just choose a fixed one in the test's for-loop. The "unit" test for this is the endpoint routing, not the search engine service, so you shouldn't have to vary the service request/responses. I would expect the querying logic unit to be tested in a different test suite (which is done in the other test file).

Reply:
I just removed the aforementioned fields from the test cases @anniefu. Although, I am not completely sure if I got it right hahaha.

Reply:
Yep, that's the idea! Could you do the same for the databaseApiTests too, please?

Reply:
For sure 👍